pkg/helper/machinepools/helpers.go

```diff
@@ -230,26 +234,59 @@ func CreateNodeDrainGracePeriodBuilder(nodeDrainGracePeriod string) (*cmv1.Value
 	}
 
 	nodeDrainParsed := strings.Split(nodeDrainGracePeriod, " ")
-	if len(nodeDrainParsed) > 2 {
-		return nil, fmt.Errorf("Invalid value for node drain grace period. Expects only the duration and " +
-			"the unit (minute|minutes|hour|hours).")
-	}
 	nodeDrainValue, err := strconv.ParseFloat(nodeDrainParsed[0], commonUtils.MaxByteSize)
 	if err != nil {
 		return nil, fmt.Errorf("Invalid time for the node drain grace period: %s", err)
 	}
 
 	// Default to minutes if no unit is specified
 	if len(nodeDrainParsed) > 1 {
-		if nodeDrainParsed[1] != "hours" && nodeDrainParsed[1] != "hour" &&
-			nodeDrainParsed[1] != "minutes" && nodeDrainParsed[1] != "minute" {
-			return nil, fmt.Errorf("Invalid unit for the node drain grace period. Valid value is `minute|minutes|hour|hours`")
-		}
-		if nodeDrainParsed[1] == "hours" || nodeDrainParsed[1] == "hour" {
+		if nodeDrainParsed[1] == nodeDrainUnitHours || nodeDrainParsed[1] == nodeDrainUnitHour {
 			nodeDrainValue = nodeDrainValue * 60
 		}
 	}
 
 	valueBuilder.Value(nodeDrainValue).Unit("minutes")
 	return valueBuilder, nil
 }
+
+func ValidateNodeDrainGracePeriod(val interface{}) error {
+	nodeDrainGracePeriod := val.(string)
+	if nodeDrainGracePeriod == "" {
+		return nil
+	}
+
+	nodeDrainParsed := strings.Split(nodeDrainGracePeriod, " ")
+	if len(nodeDrainParsed) > 2 {
+		return fmt.Errorf("Expects format to include the duration and " +
+			"the unit (minute|minutes|hour|hours).")
+	}
+	nodeDrainValue, err := strconv.ParseInt(nodeDrainParsed[0], 10, 64)
+	if err != nil {
+		return fmt.Errorf("Invalid value '%s', must be an integer.",
+			nodeDrainParsed[0])
+	}
+
+	// Default to minutes if no unit is specified
+	if len(nodeDrainParsed) > 1 {
+		if nodeDrainParsed[1] != nodeDrainUnitHours && nodeDrainParsed[1] != nodeDrainUnitHour &&
+			nodeDrainParsed[1] != "minutes" && nodeDrainParsed[1] != "minute" {
```

Comment:
@philipwu08 For consistency, can we make `minutes` and `minute` a `const` as well, like you've done with `hour` and `hours` 👍